Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
My scene only has one object (aside from the Main camera and directional light)

It's this:
![image](https://cloud.githubusercontent.com/assets/15933792/18886910/d2923dae-84a6-11e6-9962-867ef863ba49.png)

the listeners all look like:
![image](https://cloud.githubusercontent.com/assets/15933792/18886929/e893f91c-84a6-11e6-9e7c-c4249db505a4.png)

Here's the scripts, I just added an interface:
[scriptszippedup.zip](https://github.com/Microsoft/HoloToolkit-Unity/files/496407/scriptszippedup.zip)